src/System.IO.Compression.Brotli/System/IO/Compression/Brotli/Decoder.cs

```diff
@@ -10,9 +10,9 @@ namespace System.IO.Compression
 {
     internal sealed class Decoder
     {
-        internal IntPtr State = IntPtr.Zero;
-        internal BrotliDecoderResult LastDecoderResult = BrotliDecoderResult.NeedsMoreInput;
-        internal MemoryStream BufferStream;
+        internal IntPtr State { get; private set; }
+        internal BrotliDecoderResult LastDecoderResult { get; set; }
+        internal MemoryStream _bufferStream;
```

Comment:
This should probably be a private set property as well. Also, consider changing MemoryStream to Stream for the type. We shouldn't be leaking implementation details.